### PR TITLE
`sage -package dependencies`

### LINKED ISSUE #33860: sage-package: Add commands "dependencies", "trees"
<div id="comment:0"></div>

to simplify bootstrapping and configure code

Component: **refactoring**

_Issue created by migration from https://trac.sagemath.org/ticket/33860_

### PR DESCRIPTION
**Pull request by mkoeppe**
We add a new command `sage --package dependencies`.
Like the command `sage --package properties` introduced in #37018, it can be used to eliminate direct references to `build/pkgs/` from various scripts and reduce code duplication in reading SPKG metadata.

Here we only make such changes in `sage-spkg-info`.
The only user-visible change is that the listed dependencies are now sorted alphabetically; this hides [subtle nuances that may be expressed in the sort order of some `dependencies` files](https://github.com/sagemath/sage/pull/36878#issuecomment-1876426258).

- Closes #33860

Split out from:
- #36740

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by kwankyu** [comment]
I get
```bash
$ sage --package properties furo singular
furo:
        path:                        /Users/kwankyu/GitHub/sage-dev/build/pkgs/furo
        version_with_patchlevel:     2022.9.29
        type:                        standard
        source:                      wheel
        trees:                       SAGE_VENV
singular:
        path:                        /Users/kwankyu/GitHub/sage-dev/build/pkgs/singular
        version_with_patchlevel:     4.3.2p8
        type:                        standard
        source:                      normal
        trees:                       SAGE_LOCAL

$ sage --package dependencies furo singular
$(MP_LIBRARY)
$(PYTHON)
$(PYTHON_TOOLCHAIN)
beautifulsoup4
cddlib
flint
mpfr
ntl
pygments
readline
sphinx
sphinx_basic_ng

$ sage --package dependencies --format=shell furo singular
order_only_deps_furo='$(PYTHON_TOOLCHAIN) $(PYTHON)'
optional_deps_furo=''
runtime_deps_furo='beautifulsoup4 sphinx pygments sphinx_basic_ng'
check_deps_furo=''
order_only_deps_singular=''
optional_deps_singular=''
runtime_deps_singular='$(MP_LIBRARY) ntl flint readline mpfr cddlib'
check_deps_singular=''
```
It seems that the output of `sage --package dependencies furo singular` should be improved.

**Comment by mkoeppe** [comment]
> It seems that the output of `sage --package dependencies furo singular` should be improved.

I've made this change and also moved the whole formatting of dependencies for text and RST into sage_bootstrap.

**Comment by kwankyu** [comment]
Before
```
$ build/bin/sage-spkg-info furo

Dependencies
------------


Version Information
-------------------


Equivalent System Packages
--------------------------

(none known)

```
After
```
$ build/bin/sage-spkg-info furo
sage-spkg-info: unknown package furo
```

File: bootstrap [hunk 1]
```diff
@@ -28,6 +28,8 @@ export SAGE_ROOT
 cd "$SAGE_ROOT"
 
 export PATH="$SAGE_ROOT/build/bin:$PATH"
+# Faster than calling sage-package
+export SAGE_PACKAGE="env PYTHONPATH=$SAGE_ROOT/build $(sage-bootstrap-python -c 'import sys; print(sys.executable)') -m sage_bootstrap.cmdline"
```
Comment (kwankyu) [inline bootstrap]: Is the speedup really significant enough to justify the regression in readability of the code as in 
```diff
-    for pkgname in $(sage-package list :standard:); do
+    for pkgname in $(${SAGE_PACKAGE-sage-package} list :standard:); do
```
Why do you need the speedup?

**Comment by mkoeppe** [comment]
> ```
> $ build/bin/sage-spkg-info furo
> sage-spkg-info: unknown package furo
> ```

Well, the tools in `build/bin` are not consistent regarding whether they work if build/bin is not in PATH.

File: bootstrap (hunk 1 above)
Comment (mkoeppe) [reply bootstrap]: I'll check whether it's still needed after the last change.

**Comment by kwankyu** [comment]
OK.

File: bootstrap (hunk 1 above)
Comment (mkoeppe) [reply bootstrap]: Not needed any more, I've taken it out.

Comment (kwankyu) [reply bootstrap]: Not yet?

Comment (mkoeppe) [reply bootstrap]: There was one holdout. Fixed now.

**Comment by kwankyu** [comment]
Because of unnecessary indentation in "Dependencies" section as seen here 
```
$ ./sage-spkg-info furo
furo: A clean customizable Sphinx documentation theme

...

Type
----

standard


Dependencies
------------

                - $(PYTHON)
                - $(PYTHON_TOOLCHAIN)
                - beautifulsoup4
                - pygments
                - sphinx
                - sphinx_basic_ng

Version Information
-------------------

...
```
we have
Screen Shot 2024-02-21 at 2 10 54 PM
instead of (better looking)
Screen Shot 2024-02-21 at 2 12 50 PM

**Comment by mkoeppe** [comment]
Thanks! Let's see if I got it right now

**Review by kwankyu — APPROVED** [review]
Thanks. LGTM.

**Comment by kwankyu** [comment]
It works well.

**Comment by mkoeppe** [comment]
Thanks for the careful review!

**Pull request merged**